firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/send/DataTransportCrashlyticsReportSenderTest.java

```diff
@@ -14,82 +14,96 @@
 
 package com.google.firebase.crashlytics.internal.send;
 
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.doAnswer;
 import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.verify;
 
 import androidx.test.runner.AndroidJUnit4;
-import com.google.android.datatransport.Event;
 import com.google.android.datatransport.Transport;
+import com.google.android.datatransport.TransportScheduleCallback;
+import com.google.android.gms.tasks.Task;
 import com.google.firebase.crashlytics.internal.model.CrashlyticsReport;
-import java.util.ArrayList;
-import java.util.List;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.Mock;
 import org.mockito.MockitoAnnotations;
+import org.mockito.stubbing.Answer;
 
 @RunWith(AndroidJUnit4.class)
 public class DataTransportCrashlyticsReportSenderTest {
 
   @Mock private Transport<CrashlyticsReport> mockTransport;
 
-  @Mock private SendCallback<CrashlyticsReport> mockCallback;
-
   private DataTransportCrashlyticsReportSender reportSender;
 
   @Before
   public void setUp() throws Exception {
     MockitoAnnotations.initMocks(this);
-    reportSender = new DataTransportCrashlyticsReportSender(mockTransport, mockCallback);
+    reportSender = new DataTransportCrashlyticsReportSender(mockTransport);
   }
 
   @Test
   public void testSendReportsSuccessful() {
-    doAnswer(
-            (i) -> {
-              final Event<CrashlyticsReport> event = i.getArgument(0);
-              mockCallback.onSendComplete(event.getPayload(), null);
-              return null;
-            })
-        .when(mockTransport)
-        .schedule(any(), any());
+    doAnswer(callbackAnswer(null)).when(mockTransport).schedule(any(), any());
 
     final CrashlyticsReport report1 = mock(CrashlyticsReport.class);
     final CrashlyticsReport report2 = mock(CrashlyticsReport.class);
 
-    final List<CrashlyticsReport> reports = new ArrayList<>();
-    reports.add(report1);
-    reports.add(report2);
+    final Task<CrashlyticsReport> send1 = reportSender.sendReport(report1);
+    final Task<CrashlyticsReport> send2 = reportSender.sendReport(report2);
 
-    reportSender.sendReports(reports);
-    verify(mockCallback).onSendComplete(report1, null);
-    verify(mockCallback).onSendComplete(report2, null);
+    assertTrue(send1.isSuccessful());
```

Comment:
It would be better to call `Tasks.await(send1);` before checking the result of the task. Even if the test happens to resolve the task synchronously right now, there are lots of reasons why it might become async in the future, and that way the tests will keep working as expected. Same for similar usages below.

Reply:
Good call. I'll add that. Thanks!